app/src/main/java/com/ucm/informatica/spread/Fragments/HomeFragment.java

```diff
@@ -46,31 +34,49 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        View view = inflater.inflate(R.layout.fragment_home, container, false);
-        initView(view);
+        view = inflater.inflate(R.layout.fragment_home, container, false);
+        initView();
         homeFragmentPresenter = new HomeFragmentPresenter(this, this);
         setupListeners();
         return view;
     }
-    private void initView(View v) {
-        nameText = v.findViewById(R.id.nameText);
-        nameEditText = v.findViewById(R.id.nameEditText);
-        saveButton = v.findViewById(R.id.saveNameButton);
-        loadButton = v.findViewById(R.id.loadNameButton);
+    private void initView() {
+        nameText = view.findViewById(R.id.nameText);
+        helpButton = view.findViewById(R.id.helpButton);
+        cameraButton = view.findViewById(R.id.addAdvertisingButton);
     }
 
     private void setupListeners(){
-        saveButton.setOnClickListener(view -> homeFragmentPresenter.saveData(nameEditText.getText().toString()));
-        loadButton.setOnClickListener(view -> homeFragmentPresenter.loadData());
+        helpButton.setOnClickListener(view -> {
+                Location location = ((MainTabActivity) getActivity()).getLocation();
+                if(location != null) {
```

Comment:
when SOS button is pressed, info will be sent to ethereum and we need the current location